server/core/core.steps.ts

```diff
@@ -37,90 +37,65 @@ When(
-          'x-download-options': 'noopen', // for older browsers, do not trigger downloads
-          'x-content-type-options': 'nosniff', // prevent mime checking/sniffing
-          'x-permitted-cross-domain-policies': 'none', // defines a location where a policy configuration could be retrieved for some clients to access content across multiple domains. Disabled here.
-          'content-security-policy':
-            "default-src 'self';base-uri 'self';block-all-mixed-content;font-src 'self' https: data:;frame-ancestors 'self';img-src 'self' data:;object-src 'none';script-src 'self';script-src-attr 'none';style-src 'self' https: 'unsafe-inline';upgrade-insecure-requests",
-          'x-dns-prefetch-control': 'off', // CSP to help prevent cross site scripting by restricting all content to come from 'here' (self)
-          'expect-ct': 'max-age=0', // allow the checking of/confirm certificates used are used correctly
-          'referrer-policy': 'no-referrer', // remove the previously visited/linked location
-        };
+    const chainedRequest = request.expect(
+      (res) => res.headers['x-powered-by'] === undefined
+    );
+    // object with all expected headers and values
+    const headerExpectedValue = {
+      'x-xss-protection': '0', // disable XSS auditor in browser, superseded by content-security-policy
+      'x-frame-options': 'SAMEORIGIN', // prevents clickjacking - only allows the rendering/use of returned content in a frame with the same origin
+      'strict-transport-security': 'max-age=15552000; includeSubDomains', // tell the browser to prefer/use HTTPS over HTTP where possible
+      'x-download-options': 'noopen', // for older browsers, do not trigger downloads
+      'x-content-type-options': 'nosniff', // prevent mime checking/sniffing
+      'x-permitted-cross-domain-policies': 'none', // defines a location where a policy configuration could be retrieved for some clients to access content across multiple domains. Disabled here.
+      'content-security-policy':
+        "default-src 'self';base-uri 'self';block-all-mixed-content;font-src 'self' https: data:;frame-ancestors 'self';img-src 'self' data:;object-src 'none';script-src 'self';script-src-attr 'none';style-src 'self' https: 'unsafe-inline';upgrade-insecure-requests",
+      'x-dns-prefetch-control': 'off', // CSP to help prevent cross site scripting by restricting all content to come from 'here' (self)
+      'expect-ct': 'max-age=0', // allow the checking of/confirm certificates used are used correctly
+      'referrer-policy': 'no-referrer', // remove the previously visited/linked location
+    };
 
-        Object.entries(headerExpectedValue).forEach(([key, value]) =>
-          expect(headers[key]).toEqual(value)
-        );
-      },
-      (err) => {
-        throw err;
-      }
+    return Object.entries(headerExpectedValue).reduce(
```

Comment:
This is a lot more subtle/less readable than the prior version IMO. Can a comment be added to explain what this is doing/how the `chainedRequest` is being invoked?

Reply:
I'll revert - if it needs a comment then the efficiency isn't worth the readable impact